runtime/mixed.inl

```diff
@@ -1579,6 +1745,10 @@ inline bool mixed::is_reference_counter(ExtraRefCnt ref_cnt_value) const noexcep
       return as_string().is_reference_counter(ref_cnt_value);
     case type::ARRAY:
       return as_array().is_reference_counter(ref_cnt_value);
+    case type::OBJECT:
+      // TODO fix later to remove const cast
+      // Have to change codegen such that get_refcnt has const qualifier
+      return const_cast<may_be_mixed_base*>(as_object())->get_refcnt() == ref_cnt_value;
```

Comment:
It's very dangerous to have `const_cast`. I'd prefer to get rid of it in this PR unless it takes too much efforts

Reply:
fixed

Reply:
Fixed